### PR TITLE
Fix AttributeError when creating AnalysisSpec with results range via JSONAPI

### LINKED ISSUE #2428: failed to create an analysisspec with ResultsRange field by jsonapi
## Description
create an analysisspec by JSONAPI with ResultsRange field by jsonapi, 
POST Method, data =
{
    "title": "spec_name",
    "description": "desc. of analysisspec",
    "ResultsRange": [

            {
                  "keyword": "kw-01",
                  "min": "700",
                  "max": "1000"
            },
            {
                  "keyword": "kw-02",
                  "min": "600",
                  "max": "900",                  
            }

      ],
    "portal_type": "AnalysisSpec",
    "SampleType": uid_sampletype",
    "parent_path": "/senaite/bika_setup/bika_analysisspecs"
}
the create cannot success. 

<<DESCRIPTION OF THE PROBLEM>>

## Reproduction guide

<<REPRODUCTION STEPS>>

## Observed behaviour
feedback message " 'NoneType' object has no attribute 'get' "
but when the data remove the "ResultsRange", the create is success.

<<DESCRIPTION OF THE OBSERVED BEHAVIOUR>>

## Expected behaviour
could create analysisspec by JSONAPI and set "ResultsRange" field at the same time.
<<DESCRIPTION OF THE EXPECTED BEHAVIOUR>>

## Versions

- OS: Linux

- senaite.core: `master`
- senaite.impress: `master`
- senaite.core.listing: `master`
- senaite.core.supermodel: `master`

## Traceback
```
Traceback (most recent call last):
  File "/home/senaite/buildout-cache/eggs/cp27mu/plone.jsonapi.core-0.7.0-py2.7.egg/plone/jsonapi/core/browser/decorators.py", line 23, in decorator
    return f(*args, **kwargs)
  File "/home/senaite/buildout-cache/eggs/cp27mu/plone.jsonapi.core-0.7.0-py2.7.egg/plone/jsonapi/core/browser/api.py", line 57, in to_json
    return self.dispatch()
  File "/home/senaite/buildout-cache/eggs/cp27mu/plone.jsonapi.core-0.7.0-py2.7.egg/plone/jsonapi/core/browser/api.py", line 51, in dispatch
    return router(self.context, self.request, path)
  File "/home/senaite/buildout-cache/eggs/cp27mu/plone.jsonapi.core-0.7.0-py2.7.egg/plone/jsonapi/core/browser/router.py", line 150, in __call__
    return self.view_functions[endpoint](context, request, **values)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.jsonapi-2.4.0-py2.7.egg/senaite/jsonapi/v1/routes/content.py", line 105, in action
    items = action_func(portal_type=portal_type, uid=uid)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.jsonapi-2.4.0-py2.7.egg/senaite/jsonapi/api.py", line 158, in create_items
    obj = create_object(container, portal_type, **record)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.jsonapi-2.4.0-py2.7.egg/senaite/jsonapi/api.py", line 1419, in create_object
    update_object_with_data(obj, data)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.jsonapi-2.4.0-py2.7.egg/senaite/jsonapi/api.py", line 1495, in update_object_with_data
    invalid = validate_object(content, record)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.jsonapi-2.4.0-py2.7.egg/senaite/jsonapi/api.py", line 1524, in validate_object
    return obj.validate(data=data)
  File "/home/senaite/buildout-cache/eggs/cp27mu/Products.Archetypes-1.16.6-py2.7.egg/Products/Archetypes/BaseObject.py", line 517, in validate
    errors=errors, data=data, metadata=metadata)
  File "/home/senaite/buildout-cache/eggs/cp27mu/Products.Archetypes-1.16.6-py2.7.egg/Products/Archetypes/Schema/__init__.py", line 629, in validate
    REQUEST=REQUEST)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.core-2.4.1-py2.7.egg/senaite/core/browser/fields/records.py", line 173, in validate
    ), None)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.core-2.4.1-py2.7.egg/senaite/core/browser/fields/records.py", line 169, in <genexpr>
    result
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.core-2.4.1-py2.7.egg/senaite/core/browser/fields/records.py", line 165, in <genexpr>
    for record
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.core-2.4.1-py2.7.egg/senaite/core/browser/fields/record.py", line 390, in validate
    res = self.validate_validators(value, instance, errors, **kwargs)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.core-2.4.1-py2.7.egg/senaite/core/browser/fields/record.py", line 407, in validate_validators
    **kwargs
  File "/home/senaite/buildout-cache/eggs/cp27mu/Products.validation-2.1.3-py2.7.egg/Products/validation/chain.py", line 137, in __call__
    result = validator(value, *args, **kwargs)
  File "/home/senaite/buildout-cache/eggs/cp27mu/senaite.core-2.4.1-py2.7.egg/bika/lims/validators.py", line 845, in __call__
    service_uids = request.get("uids", [])
AttributeError: 'NoneType' object has no attribute 'get'
<<BACKTRACE IF RELEVANT>>
```

### PR DESCRIPTION
**Pull request by fengyundeng**
## Description of the issue/feature this PR addresses

Linked issue: https://github.com/senaite/senaite.core/issues/

## Current behavior before PR
in src/bika/lims/validators.py, line 818
          request = kwargs.get("REQUEST", {})
request is None, when create analysisspec by JSONAPI.
then, in src/bika/lims/validators.py, line 829
          service_uids = request.get("uids", [])
raise an error, 'NoneType' object has no attribute 'get'

## Desired behavior after PR is merged
fix the bug when "request" is None.
the analysisspec with "ResultsRange" by JSONAPI created work fine.

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by fengyundeng** [comment]
I am new with Senaite.LIMS. When i want to create analysis spec by JSONAPI, i found the bug "request" is None and raise an error 'NoneType' object has no attribute 'get'. the fix is work fine for me. But i didn't go very deeper in code (as I'm new one). So please check the fix is in the way properly.

File: src/bika/lims/validators.py [hunk 1]
```diff
@@ -816,6 +816,8 @@ class AnalysisSpecificationsValidator:
     def __call__(self, value, *args, **kwargs):
         instance = kwargs["instance"]
         request = kwargs.get("REQUEST", {})
+        if request is None:
+            request = {}
```
Comment (xispa) [inline src/bika/lims/validators.py]: You can simplify with:

```python
request = kwargs.get("REQUEST") or {}
```

**Review by xispa — CHANGES_REQUESTED** [review]
Thanks @fengyundeng , please add an entry to the changelog as well: https://github.com/senaite/senaite.core/blob/2.x/CHANGES.rst

File: src/bika/lims/validators.py (hunk 1 above)
Comment (fengyundeng) [reply src/bika/lims/validators.py]: thanks @xispa for the nice feedback.

**Comment by fengyundeng** [comment]
sorry for caused the confusion.
i'm not familiar with pull process.

**Review by xispa — APPROVED** [review]

**Pull request merged**